scylla/src/policies/host_listener.rs

```diff
@@ -0,0 +1,81 @@
+//! Host listeners can subscribe to events regarding hosts.
+//!
+//! This includes events when a host is added or removed from the cluster,
+//! as well as when a host is marked as up or down.
+
+use std::net::SocketAddr;
+
+use uuid::Uuid;
+
+/// Context provided to [HostListener] callbacks.
+#[derive(Debug)]
+pub struct HostEventContext {
+    pub(crate) host_id: Uuid,
+    pub(crate) addr: SocketAddr,
+}
+
+impl HostEventContext {
+    /// ID of the host related to the event.
+    pub fn host_id(&self) -> Uuid {
+        self.host_id
+    }
+
+    /// Address of the host related to the event.
+    pub fn addr(&self) -> &SocketAddr {
+        &self.addr
+    }
+}
+
+/// Kind of host event being signalled by [HostListener].
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum HostEvent {
+    /// Signalled when the driver learns that a new host is added to the cluster.
+    ///
+    /// This happens in the following cases:
+    /// 1. When the control connection discovers a new peer.
+    /// 2. When a CQL event notifies about a new node joining the cluster.
+    Added,
+
+    /// Signalled when the driver learns that a known host is removed from the cluster.
+    ///
+    /// This happens in the following cases:
+    /// 1. When the control connection discovers that a known peer is no longer in the cluster.
+    /// 2. When a CQL event notifies about a new node leaving the cluster.
+    /// 3. When a CQL event notifies about a known node being added. Then this event is fired
+    ///    right before the `host_added` event.
+    Removed,
```

Comment:
Huh, so this comment now tells me that events really are taken into consideration. Interesting, I definitely need to read the implementation.

Reply:
This is as well outdated.